win32ss/gdi/ntgdi/freetype.c

```diff
@@ -79,30 +259,22 @@ static UNICODE_STRING g_FontRegPath =
 static PFAST_MUTEX      g_FreeTypeLock;
 
 static RTL_STATIC_LIST_HEAD(g_FontListHead);
-static PFAST_MUTEX      g_FontListLock;
 static BOOL             g_RenderingEnabled = TRUE;
 
-#define IntLockGlobalFonts() \
-    ExEnterCriticalRegionAndAcquireFastMutexUnsafe(g_FontListLock)
-
-#define IntUnLockGlobalFonts() \
-    ExReleaseFastMutexUnsafeAndLeaveCriticalRegion(g_FontListLock)
-
-#define ASSERT_GLOBALFONTS_LOCK_HELD() \
-    ASSERT(g_FontListLock->Owner == KeGetCurrentThread())
-
-#define IntLockFreeType() \
-    ExEnterCriticalRegionAndAcquireFastMutexUnsafe(g_FreeTypeLock)
-
-#define IntUnLockFreeType() \
-    ExReleaseFastMutexUnsafeAndLeaveCriticalRegion(g_FreeTypeLock)
-
 #define ASSERT_FREETYPE_LOCK_HELD() \
     ASSERT(g_FreeTypeLock->Owner == KeGetCurrentThread())
 
 #define ASSERT_FREETYPE_LOCK_NOT_HELD() \
     ASSERT(g_FreeTypeLock->Owner != KeGetCurrentThread())
 
+#define IntLockFreeType() do { \
+    ASSERT_FREETYPE_LOCK_NOT_HELD(); \
+    ExEnterCriticalRegionAndAcquireFastMutexUnsafe(g_FreeTypeLock); \
+} while (0)
+
+#define IntUnLockFreeType() \
+    ExReleaseFastMutexUnsafeAndLeaveCriticalRegion(g_FreeTypeLock)
```

Comment:
You could do a `ASSERT_FREETYPE_LOCK_HELD();` before.

Reply:
Done.